### PR TITLE
BEAM-78 - First step of the renaming: Maven coordonates

### PR DESCRIPTION
**Pull request by jbonofre**

### CODE REVIEW

File: examples/pom.xml [hunk 1]
```diff
@@ -1,38 +1,35 @@
...
+
   <modelVersion>4.0.0</modelVersion>
 
   <parent>
-    <groupId>com.google.cloud.dataflow</groupId>
-    <artifactId>google-cloud-dataflow-java-sdk-parent</artifactId>
-    <version>1.5.0-SNAPSHOT</version>
+    <groupId>org.apache.beam</groupId>
+    <artifactId>parent</artifactId>
+    <version>1.5.0-incubating-SNAPSHOT</version>
```
Comment (kennknowles) [inline examples/pom.xml]: Let us use 0.0.1 to indicate the fresh start and expected churn. It is also quite handy to have a build number (perhaps YYYYmmddMMHHSS) associated with snapshots.

**Comment by davorbonaci** [comment]
I'll take a look shortly.

R: @davorbonaci

File: examples/pom.xml (hunk 1 above)
Comment (mxm) [inline examples/pom.xml]: Wouldn't it make sense to keep the Dataflow version number? Essentially, this is a fork and we're not starting from zero.

Comment (kennknowles) [inline examples/pom.xml]: Even though I suggested it, I don't feel too strongly about using `0.0.1`, I just want to maintain good [semantic versioning](http://semver.org). So another good choice is `2.0.0-SNAPSHOT`.

For getting a version to someone before the first stable release, we can offer a `0.x.y` release or a `2.0.0-SNAPSHOT` pre-release, with similar assurance of quality and compatibility. Whatever works best for everyone and the incubating process.

I notice that there is a suggestion on the [technical vision doc](https://docs.google.com/document/d/1UyAeugHxZmVlQ5cEWo_eOPgXNQA1oD-rGooWOSwAqh8/edit?usp=sharing) that we have a release plan. That would drive the decision.

Comment (mxm) [inline examples/pom.xml]: @kennknowles I'm all for sensible semantic versioning but the document only states the versioning for releases. In the Maven world, `SNAPSHOT` indicates a non-released minor version. Usually, it doesn't carry the "patch" part of the release version. The reasoning behind that is to snapshot the minor version during development and backport patches into patch releases.

Minor Release: 0.1.0
Patch release: 0.1.1
Unreleased minor version: 0.2-SNAPSHOT
Unreleased major version: 1.0-SNAPSHOT
Major release: 1.0.0

The design doc suggests to start from `0.x.y`. So I guess it makes sense to stick to the initial proposal then and make it `0.1-SNAPSHOT`. Perhaps we should move the discussion to the mailing list :)

Comment (jbonofre) [inline examples/pom.xml]: Agree with Max. 0.1-incubating-SNAPSHOT makes sense. We should discuss this on the mailing list anyway.

Comment (kennknowles) [inline examples/pom.xml]: Sounds great.

Were we going to get commits@ subscribed to the firehose of all comments on all PRs? I think that would be "helpful" for spreading knowledge about developments, and make it easy to pull things from commits@ to dev@

Comment (davorbonaci) [inline examples/pom.xml]: (see my comment on the dev mailing list)

File: examples/pom.xml [hunk 2]
```diff
@@ -1,38 +1,35 @@
...
-  ~ Unless required by applicable law or agreed to in writing, software
-  ~ distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
-  ~ WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
-  ~ License for the specific language governing permissions and limitations under
-  ~ the License.
-  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~-->
-<project xmlns="http://maven.apache.org/POM/4.0.0"
-         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (davorbonaci) [inline examples/pom.xml]: What's the rationale being wrapping change? It is easier to read 3 shorter lines, than one really long one.

File: examples/pom.xml [hunk 3]
```diff
@@ -1,38 +1,35 @@
...
+
+  <!--
+
+      Licensed to the Apache Software Foundation (ASF) under one or more
```
Comment (davorbonaci) [inline examples/pom.xml]: What's the rationale behind the change in ordering of the <project> tag and license comment? We want it on top of the file, no? If so, it can be above <project>.

File: examples/pom.xml [hunk 4]
```diff
@@ -1,38 +1,35 @@
...
+  -->
+
   <modelVersion>4.0.0</modelVersion>
 
   <parent>
-    <groupId>com.google.cloud.dataflow</groupId>
-    <artifactId>google-cloud-dataflow-java-sdk-parent</artifactId>
-    <version>1.5.0-SNAPSHOT</version>
+    <groupId>org.apache.beam</groupId>
+    <artifactId>parent</artifactId>
```
Comment (davorbonaci) [inline examples/pom.xml]: +1 on the artifactId change. Like it!

File: examples/pom.xml [hunk 5]
```diff
@@ -1,38 +1,35 @@
...
 
-  <groupId>com.google.cloud.dataflow</groupId>
-  <artifactId>google-cloud-dataflow-java-examples-all</artifactId>
-  <name>Google Cloud Dataflow Java Examples - All</name>
-  <description>Google Cloud Dataflow Java SDK provides a simple, Java-based
-    interface for processing virtually any size data using Google cloud
-    resources. This artifact includes all Dataflow Java SDK
-    examples.</description>
-  <url>http://cloud.google.com/dataflow</url>
+  <artifactId>examples</artifactId>
```
Comment (davorbonaci) [inline examples/pom.xml]: Can we call this "java-examples-all"? This gives us a way out if one day we decide something like:
* separate the module into pieces; then we can have "examples-part1", "examples-part2", and "examples-all" still pulling everything in.
* publish non-java examples to Maven.

File: examples/pom.xml [hunk 6]
```diff
@@ -1,38 +1,35 @@
...
+  <name>Apache Beam :: Examples</name>
```
Comment (davorbonaci) [inline examples/pom.xml]: Add <description>?

File: maven-archetypes/examples/pom.xml [hunk 7]
```diff
@@ -1,37 +1,36 @@
...
-  ~ Unless required by applicable law or agreed to in writing, software
-  ~ distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
-  ~ WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
-  ~ License for the specific language governing permissions and limitations under
-  ~ the License.
-  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~-->
-<project xmlns="http://maven.apache.org/POM/4.0.0"
-         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (davorbonaci) [inline maven-archetypes/examples/pom.xml]: same comments as above: wrapping of `<project>`, ordering of `<project>` and the license header.

File: maven-archetypes/examples/pom.xml [hunk 8]
```diff
@@ -1,37 +1,36 @@
...
     <relativePath>../../pom.xml</relativePath>
   </parent>
 
-  <groupId>com.google.cloud.dataflow</groupId>
-  <artifactId>google-cloud-dataflow-java-archetypes-examples</artifactId>
-  <name>Google Cloud Dataflow Java SDK - Examples Archetype</name>
-  <description>A Maven Archetype to create a project containing all the
-    example pipelines from the Google Cloud Dataflow Java SDK.</description>
-  <url>http://cloud.google.com/dataflow</url>
+  <groupId>org.apache.beam.archetypes</groupId>
```
Comment (davorbonaci) [inline maven-archetypes/examples/pom.xml]: Why do we need a separate groupId here? Not sure if it buys us something, and we likely need more paperwork to own another group in Maven.

File: maven-archetypes/examples/pom.xml [hunk 9]
```diff
@@ -1,37 +1,36 @@
...
+  <artifactId>examples</artifactId>
```
Comment (davorbonaci) [inline maven-archetypes/examples/pom.xml]: "java-archetypes-examples" perhaps? Same reasoning as above.

File: maven-archetypes/examples/pom.xml [hunk 10]
```diff
@@ -1,37 +1,36 @@
...
+  <name>Apache Beam :: Archetypes :: Examples</name>
```
Comment (davorbonaci) [inline maven-archetypes/examples/pom.xml]: add `<description>`?

File: maven-archetypes/examples/pom.xml [hunk 11]
```diff
@@ -1,37 +1,36 @@
...
   <modelVersion>4.0.0</modelVersion>
 
   <parent>
-    <groupId>com.google.cloud.dataflow</groupId>
-    <artifactId>google-cloud-dataflow-java-sdk-parent</artifactId>
-    <version>1.5.0-SNAPSHOT</version>
+    <groupId>org.apache.beam</groupId>
+    <artifactId>parent</artifactId>
+    <version>1.5.0-incubating-SNAPSHOT</version>
     <relativePath>../../pom.xml</relativePath>
```
Comment (davorbonaci) [inline maven-archetypes/examples/pom.xml]: We could perhaps have a parent pom for all archetypes.

File: maven-archetypes/starter/pom.xml [hunk 12]
```diff
@@ -1,37 +1,36 @@
...
-  ~ Unless required by applicable law or agreed to in writing, software
-  ~ distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
-  ~ WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
-  ~ License for the specific language governing permissions and limitations under
-  ~ the License.
-  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~-->
-<project xmlns="http://maven.apache.org/POM/4.0.0"
-         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (davorbonaci) [inline maven-archetypes/starter/pom.xml]: same comments as above: wrapping of <project>, ordering of <project> and the license header.

File: maven-archetypes/starter/pom.xml [hunk 13]
```diff
@@ -1,37 +1,36 @@
...
     <relativePath>../../pom.xml</relativePath>
   </parent>
 
-  <groupId>com.google.cloud.dataflow</groupId>
-  <artifactId>google-cloud-dataflow-java-archetypes-starter</artifactId>
-  <name>Google Cloud Dataflow Java SDK - Starter Archetype</name>
-  <description>A Maven archetype to create a simple starter pipeline to
-    get started using the Google Cloud Dataflow Java SDK. </description>
-  <url>http://cloud.google.com/dataflow</url>
+  <groupId>org.apache.beam.archetypes</groupId>
```
Comment (davorbonaci) [inline maven-archetypes/starter/pom.xml]: Same comments as above:
* why archetypes groupid?
* add description.
* java-archetypes-starter

File: pom.xml [hunk 14]
```diff
@@ -1,41 +1,42 @@
...
-  ~ Unless required by applicable law or agreed to in writing, software
-  ~ distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
-  ~ WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
-  ~ License for the specific language governing permissions and limitations under
-  ~ the License.
-  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~-->
-<project xmlns="http://maven.apache.org/POM/4.0.0"
-         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (davorbonaci) [inline pom.xml]: same comments as above: wrapping of , ordering of and the license header.

File: pom.xml [hunk 15]
```diff
@@ -45,19 +46,51 @@
     </license>
   </licenses>
 
+  <scm>
```
Comment (davorbonaci) [inline pom.xml]: +1, thanks for setting all this up!

File: runners/flink/pom.xml [hunk 16]
```diff
@@ -1,54 +1,44 @@
...
+        <version>1.5.0-incubating-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
     </parent>
 
-    <artifactId>flink-runner</artifactId>
-    <version>0.3-SNAPSHOT</version>
-
-    <name>Flink Beam Runner</name>
-    <packaging>jar</packaging>
+    <groupId>org.apache.beam.runners</groupId>
```
Comment (davorbonaci) [inline runners/flink/pom.xml]: Same comment as above; I'd probably keep the group org.apache.beam for all artifacts.

File: runners/flink/pom.xml [hunk 17]
```diff
@@ -1,54 +1,44 @@
...
+    <!--
 
-    Unless required by applicable law or agreed to in writing,
-    software distributed under the License is distributed on an
-    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
-    KIND, either express or implied.  See the License for the
-    specific language governing permissions and limitations
-    under the License.
--->
-<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
```
Comment (davorbonaci) [inline runners/flink/pom.xml]: same comment as above; project wrapping, ordering of project and license.

File: runners/pom.xml [hunk 18]
```diff
@@ -1,43 +1,43 @@
...
-    Unless required by applicable law or agreed to in writing,
-    software distributed under the License is distributed on an
-    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
-    KIND, either express or implied.  See the License for the
-    specific language governing permissions and limitations
-    under the License.
--->
-<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
-        xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (davorbonaci) [inline runners/pom.xml]: same comment as above: project wrapping, ordering of project and license

File: runners/pom.xml [hunk 19]
```diff
@@ -1,43 +1,43 @@
...
-        <artifactId>google-cloud-dataflow-java-sdk-parent</artifactId>
-        <version>1.5.0-SNAPSHOT</version>
+        <groupId>org.apache.beam</groupId>
+        <artifactId>parent</artifactId>
+        <version>1.5.0-incubating-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
     </parent>
 
-    <groupId>org.apache.beam</groupId>
     <artifactId>runners</artifactId>
```
Comment (davorbonaci) [inline runners/pom.xml]: please sync to pick up Max's change to rename this runners-parent.

File: runners/pom.xml [hunk 20]
```diff
@@ -1,43 +1,43 @@
...
-        <groupId>com.google.cloud.dataflow</groupId>
-        <artifactId>google-cloud-dataflow-java-sdk-parent</artifactId>
-        <version>1.5.0-SNAPSHOT</version>
+        <groupId>org.apache.beam</groupId>
+        <artifactId>parent</artifactId>
+        <version>1.5.0-incubating-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
     </parent>
 
-    <groupId>org.apache.beam</groupId>
```
Comment (davorbonaci) [inline runners/pom.xml]: you haven't changed the groupid here. I'd definitely have runners-parent in the same group as runners-xyz.

File: sdk/pom.xml [hunk 21]
```diff
@@ -1,37 +1,36 @@
...
-  ~ Unless required by applicable law or agreed to in writing, software
-  ~ distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
-  ~ WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
-  ~ License for the specific language governing permissions and limitations under
-  ~ the License.
-  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~-->
-<project xmlns="http://maven.apache.org/POM/4.0.0"
-         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (davorbonaci) [inline sdk/pom.xml]: same comment as above: project wrapping, ordering of project and license

File: sdk/pom.xml [hunk 22]
```diff
@@ -1,37 +1,36 @@
...
   </parent>
 
-  <groupId>com.google.cloud.dataflow</groupId>
-  <artifactId>google-cloud-dataflow-java-sdk-all</artifactId>
-  <name>Google Cloud Dataflow Java SDK - All</name>
-  <description>Google Cloud Dataflow Java SDK provides a simple, Java-based
-    interface for processing virtually any size data using Google cloud
-    resources. This artifact includes entire Dataflow Java SDK.</description>
-  <url>http://cloud.google.com/dataflow</url>
+  <artifactId>sdk</artifactId>
```
Comment (davorbonaci) [inline sdk/pom.xml]: java-sdk-all, for the same reasons as above.

File: sdk/pom.xml [hunk 23]
```diff
@@ -1,37 +1,36 @@
...
+  <name>Apache Beam :: SDK</name>
```
Comment (davorbonaci) [inline sdk/pom.xml]: Java SDK :: All

**Comment by davorbonaci** [comment]
I took a quick peek; left a few comments.

Thanks JB -- this is a clear improvement.

File: examples/pom.xml (hunk 1 above)
Comment (jbonofre) [inline examples/pom.xml]: Let me update the PR with 0.1.0-incubating-SNAPSHOT then.

File: examples/pom.xml (hunk 2 above)
Comment (jbonofre) [inline examples/pom.xml]: Good point. I will use one line. By habit, I used wrapping because we had issues in the release and version plugins. But it's now fixed, so, I will use one line.

File: examples/pom.xml (hunk 3 above)
Comment (jbonofre) [inline examples/pom.xml]: Same as previous comment: in the past, the release and version plugins failed when the pom.xml didn't begin with the XML tag. Now it's fixed. So, I will update and replace the license comment on top.

File: examples/pom.xml (hunk 5 above)
Comment (jbonofre) [inline examples/pom.xml]: Good idea. I do the change.

File: examples/pom.xml (hunk 6 above)
Comment (jbonofre) [inline examples/pom.xml]: +1, I forgot, my bad. Let me fix that.

File: maven-archetypes/examples/pom.xml (hunk 8 above)
Comment (jbonofre) [inline maven-archetypes/examples/pom.xml]: I prefer to use a specific groupId, because I think it's easier for users to find what they are looking for, and avoid to have a bunch of artifacts in the same Maven repo folder. As beam will grow a lot in the future (for sure ;)), it would allow us to provide a cleaner structure IMHO.

File: maven-archetypes/examples/pom.xml (hunk 9 above)
Comment (jbonofre) [inline maven-archetypes/examples/pom.xml]: +1 I do the change

File: maven-archetypes/examples/pom.xml (hunk 10 above)
Comment (jbonofre) [inline maven-archetypes/examples/pom.xml]: +1 (I forgot, sorry about that).

File: maven-archetypes/examples/pom.xml (hunk 11 above)
Comment (jbonofre) [inline maven-archetypes/examples/pom.xml]: Agree, largely cleaner. I will do this in a second commit (folders/modules re-organization).

File: runners/pom.xml (hunk 20 above)
Comment (jbonofre) [inline runners/pom.xml]: +1 I do the change.

File: sdk/pom.xml (hunk 22 above)
Comment (jbonofre) [inline sdk/pom.xml]: Yes, it makes sense. In second commit, I will do sdk/java (in preparation for sdk/python, sdk/scala, ...).

File: sdk/pom.xml (hunk 23 above)
Comment (jbonofre) [inline sdk/pom.xml]: +1 I do the change

**Comment by jbonofre** [comment]
Updated PR including Davor's comments. I will now start the package renaming.

**Comment by davorbonaci** [comment]
There's a build break regarding archetypes. Otherwise, LGTM.

**Comment by davorbonaci** [comment]
I have fixed up this pull request to have the build pass (hopefully) and merged it.

Thanks JB!

**Comment by jbonofre** [comment]
Thanks Davor, much appreciated !
